src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -184,14 +216,27 @@ func (c *IdleTimingConn) Close() error {
 }
 
 func (c *IdleTimingConn) LocalAddr() net.Addr {
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
+
 	return c.conn.LocalAddr()
 }
 
 func (c *IdleTimingConn) RemoteAddr() net.Addr {
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
```

Comment:
Why do we need to lock here? We're not reading or writing from closed in this call right?